core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -35,10 +35,22 @@ package object config {
     ConfigBuilder(SparkLauncher.DRIVER_EXTRA_LIBRARY_PATH).stringConf.createOptional
 
   private[spark] val DRIVER_USER_CLASS_PATH_FIRST =
-    ConfigBuilder("spark.driver.userClassPathFirst").booleanConf.createWithDefault(false)
+    ConfigBuilder("spark.driver.userClassPathFirst")
+      .doc("Whether to give user-added jars precedence over Spark's own jars " +
+        "when loading classes in the driver. This feature can be used to mitigate " +
+        "conflicts between Spark's dependencies and user dependencies. " +
+        "It is currently an experimental feature. This is used in cluster mode only. ")
+      .booleanConf
+      .checkValues(Set(false, true))
+      .createWithDefault(false)
 
   private[spark] val DRIVER_MEMORY = ConfigBuilder("spark.driver.memory")
+    .doc("Amount of memory to use for the driver process, i.e. " +
+      "where SparkContext is initialized. (e.g. 1g, 2g). ")
```

Comment:
we should also mention that, if no unit is given, e.g. `500`, it means 500mb. We also need to update `configuration.md` for this conf.